core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java

```diff
@@ -151,6 +156,11 @@ public void setUp() throws IOException {
     when(taskContext.taskMetrics()).thenReturn(taskMetrics);
     when(shuffleDep.serializer()).thenReturn(serializer);
     when(shuffleDep.partitioner()).thenReturn(hashPartitioner);
+    when(taskContext.taskMemoryManager()).thenReturn(taskMemoryManager);
+
+    Random rand = new Random();
+    TaskContext$.MODULE$.setTaskContext(new TaskContextImpl(
```

Comment:
I'm a little confused about this because there's already a `taskContext` object that's mocked. why can't you just set it to that?